### PR TITLE
Ex6

### PR DESCRIPTION
**Pull request by ChipLuxury-EWA**
@barchoen2 @idanDavidi
### The requirements:
- [x] Remove `app.use(express.static(path.join(__dirname, 'dist')))` from your `server.js` file
- [x] Delete `src/server/dist` **ONLY** when your new client is up and running 
- [x] Initialize your React project in the `src` folder using [create-react-app](https://create-react-app.dev/docs/getting-started/) with the name `client`
- [x] Decompose your Todo App into components (controlled and uncontrolled)
- [x] Re-Implement the Todo App using hooks
- [x] Use [**propTypes** & **defaultProps**](https://reactjs.org/docs/typechecking-with-proptypes.html) to add type-checking to your components

### Bonus
- [ ] Add error handling for communication issues with your backend (Empty state / Loader / Something else)
- [x] Use components from the [Vibe Design System](https://github.com/mondaycom/monday-ui-react-core) (monday.com's component library)
- [ ] Use [React Router](https://reactrouter.com/docs/en/v6) to add basic routing capabilities to your app
  - [ ] Add Tabs / Navigation bar to help route between pages [Tabs example](https://style.monday.com/?path=/docs/navigation-tabs-tab--overview)
  - Page suggestions
    - [ ] Task completion statistics page - # of open tasks, total number of tasks, average tim to finish a task, etc.
    - [ ] About page - with some fun facts and profile picture

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import "monday-ui-react-core/dist/main.css";
+
+function App() {
+    const [tasks, setTasks] = React.useState([]);
+
+    React.useEffect(() => {
+        getAllTasksFromDB();
+    }, []);
+
+    const getAllTasksFromDB = async () => {
```
Comment (barcohen2) [inline client/src/App.js:23]: I think it's better that this and all the methods imported from item_Client should be inside a task list app container and not directly under App.js.
Let's keep App.js clean with only one line of rendering our application container

File: client/src/App.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    return (
+        <Router>
+        <Header />
+            <div className="main">
+                <TaskAppContainer
+                    tasksFromDB={tasks}
+                    addTask={postItem}
+                    deleteTask={deleteItem}
+                    updateItem={updateItem}
+                    updateApp={getAllTasksFromDB}
```
Comment (barcohen2) [inline client/src/App.js:38]: consider renaming updateApp to afterChangesMade for example. 
then do whatever you like inside. its more generic to use it this way.

File: client/src/components/UserInput/UserInput.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import { TextField } from "monday-ui-react-core";
+import { Add } from "monday-ui-react-core/dist/allIcons";
+import "monday-ui-react-core/dist/main.css";
+
+export const UserInput = ({
+    loading,
+    onIconClick,
+    value,
+    onChange,
+    updateApp,
```
Comment (barcohen2) [inline client/src/components/UserInput/UserInput.jsx:14]: same as in the previos comment, updateApp is not generic. afterChangeInput or something similar would be much more generic and can be used for future additional side effects that may be needed

File: client/src/components/UserInput/UserInput.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+import { TextField } from "monday-ui-react-core";
+import { Add } from "monday-ui-react-core/dist/allIcons";
+import "monday-ui-react-core/dist/main.css";
+
+export const UserInput = ({
+    loading,
+    onIconClick,
+    value,
+    onChange,
```
Comment (barcohen2) [inline client/src/components/UserInput/UserInput.jsx:13]: I dont see you passed prop named 'onChange'

File: client/src/components/UserInput/UserInput.jsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                placeholder={"Add a new task"}
+                value={task}
+                iconName={Add}
+                loading={loading}
+                onChange={(value) => {
+                    setTask(value);
+                    onChange(value);
+                }}
+                onIconClick={async () => {
+                    await onIconClick(task);
```
Comment (barcohen2) [inline client/src/components/UserInput/UserInput.jsx:29]: it's better to extract this logic to a method and call it here

**Comment by ChipLuxury-EWA** [comment]
https://user-images.githubusercontent.com/53507364/177866428-bd5bae9d-4526-4dfe-bc11-89ec68e1c367.mp4

File: client/src/App.js (hunk 1 above)
Comment (ChipLuxury-EWA) [reply client/src/App.js:23]: Moved logic to taskAppcontainr

File: client/src/App.js (hunk 2 above)
Comment (ChipLuxury-EWA) [reply client/src/App.js:38]: With redux dispatch is taking care of
will keep this note about generic way...

File: client/src/components/UserInput/UserInput.jsx (hunk 5 above)
Comment (ChipLuxury-EWA) [reply client/src/components/UserInput/UserInput.jsx:29]: I'm injecting from taskContainerApp to this comp, I practice on building a stateless components

**Pull request merged**